src/partition.ts

```diff
@@ -78,7 +78,10 @@ function partition<A extends Iterable<unknown> | AsyncIterable<unknown>, B>(
   f: (a: IterableInfer<A>) => B,
 ): (iterable: A) => ReturnPartitionType<A>;
 
-function partition<A extends Iterable<unknown> | AsyncIterable<unknown>, B>(
+function partition<
```

Comment:
But, type about partition is incorrect

I think the type implementation is weird
Changes are indispensable, and I will work on them

Reply:
```ts

function partition<
  A,
  L extends A,
  R extends A = A extends object ? ExcludeObject<A, L> : Exclude<A, L>,
>(f: (a: A) => a is L, iterable: Iterable<A>): [L[], R[]];
```
It's terrible

If this is the case, why is the 'ReturnPartition Type' implemented

Reply:
The type signature is for the case below. Please check this test case.
https://github.com/marpple/FxTS/blob/856b178c370e8016d46b15b1ee10353d89c60076/type-check/partition.test.ts#L61

You can easily check with this command: `npm run compile:check`


Could you please explain why this type is not correct?

Reply:
This type can pass through the immediate test case. However, due to the lack of consistency in the type structure, errors will be easily spat out when linked to the following types

Reply:
It would be good for me to understand if you could write an example of a problem caused by a lack of type consistency in a test code.
However, regardless of this issue, I still think there is no reason to modify 'partition' to improve the type-inferencing
 of 'groupBy' and 'indexBy'.
Please include only code for improvement of 'groupBy' and 'indexBy' in this PR, and create a separate PR for issue of 'partition' function.

Reply:
Separating the PR is also good, but there was no change in the return type in the commit that had comments. Only the as syntax was added to the return part.

Although it is a temporary measure due to the change in the return type of groupBy, it is essential. Therefore, changes to partition are inevitable in this PR.

Reply:
By the way, most of the functions provided by FxTs have the following return types:

Synchronous or asynchronous
List or iterator (ultimately iterable)
And here are the return types for partition:

[L[], R[]] (essentially, true and false parts)
[A[], A[]] (also divided into true and false)
(iterable: A) => ReturnPartitionType<A> (curried type for 2)
These scattered types result in significant maintenance costs. Is there really a need to have many divisions when they all ultimately return the same value and type?

It would be ideal and require the least maintenance cost to unify them under one type (ReturnPartitionType in this case) and consistently modify only that type.

Reply:
![image](https://user-images.githubusercontent.com/64676070/227406481-9a8e3632-c762-46e2-ad36-80bee0ddbcc6.png)
Do you want this?

Don't think that it doesn't look pretty because it's a type alias

Reply:
That's exactly what I wanted. Thank you for letting me know a good solution.